projects/core/directives/hint/hint-unstyled.component.ts

```diff
@@ -0,0 +1,48 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Directive,
+    Inject,
+    TemplateRef,
+} from '@angular/core';
+import {TUI_PARENT_ANIMATION, TuiContextWithImplicit} from '@taiga-ui/cdk';
+import {TuiPortalItem} from '@taiga-ui/core/interfaces';
+import {POLYMORPHEUS_CONTEXT, PolymorpheusComponent} from '@tinkoff/ng-polymorpheus';
+
+import {TuiHintDirective} from './hint.directive';
+
+@Component({
+    template: `
+        <ng-container
+            *polymorpheusOutlet="context.$implicit.content; context: context"
+        ></ng-container>
+    `,
+    host: {'[@tuiParentAnimation]': ''},
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    animations: [TUI_PARENT_ANIMATION],
+})
+export class TuiHintUnstyledComponent {
+    constructor(
+        @Inject(POLYMORPHEUS_CONTEXT)
+        readonly context: TuiContextWithImplicit<TuiPortalItem>,
+    ) {}
+}
+
+@Directive({
+    selector: 'ng-template[tuiHint]',
+    providers: [
+        {
+            provide: POLYMORPHEUS_CONTEXT,
+            useValue: {$implicit: {}},
+        },
+    ],
+})
+export class TuiHintUnstyledDirective<C> {
+    constructor(
+        @Inject(TemplateRef) templateRef: TemplateRef<C>,
+        @Inject(TuiHintDirective) hint: TuiHintDirective<C>,
+    ) {
+        hint.component = new PolymorpheusComponent(TuiHintUnstyledComponent);
```

Comment:
Should we also pass `INJECTOR` as the second argument ?

Reply:
I thought about it but it doesn't make any difference because it would only affect the unstyled component — the rest is a template using injector of declaration anyway.